libsolidity/interface/CompilerStack.cpp

```diff
@@ -768,6 +769,13 @@ h256 const& CompilerStack::Source::swarmHash() const
 	return swarmHashCached;
 }
 
+string const& CompilerStack::Source::ipfsUrl() const
+{
+	if (ipfsUrlCached.empty())
+		if (scanner->source().size() < 1025 * 256)
+			ipfsUrlCached = "ipfs://" + dev::ipfsHashBase58(scanner->source());
```

Comment:
We should decide which URL format to use. The IPFS project seems to prefer `/ipfs/<hash>`, but I don't think this is usable here because we just have a list of URLs instead of a key-value mapping. I have seen `ipfs://ipfs/<hash>`, but I'm not sure what the consensus is inside the ipfs project. @axic do you know more here?

Reply:
https://github.com/ipfs/go-ipfs/issues/1678#issuecomment-157478515

The canonical way seems to be `/ipfs/hash/path`, where compatibility versions exist, for example `ipfs:/ipfs/hash/path`

Reply:
See below (there are 2 years between the two comments...)